packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,410 @@
+    address this limitation, we need to extend the class and add the necessary nodes to both the serializer and parser which will be added in the
+    subsequent releases.
+-   Provide APIs to configure visibility/disabled state of toolbar buttons for a rich text editor.
+
+### Risks and Challenges
+
+-   Due to immediate requirements for comments feature from a business customer, any additional enhancements or requirements apart from whatever is
+    mentioned in this spec are deferred to future scope.
+-   Currently, we will begin by referring to the existing
+    [Interaction design workflow](https://www.figma.com/proto/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94118&scaling=min-zoom&page-id=2428%3A32954&starting-point-node-id=6280%3A94118&show-proto-sidebar=1)
+    of the comments feature. Once the visual design for these components is complete, we will then be implementing those specific changes within the defined
+    scope of development. However, we will still make use of existing nimble components such as `nimble-toggle-button` and `nimble-text-area` to maintain a
+    consistent design for the initial release.
+
+### Prior Art/Examples
+
+Sample Mockup Screenshots
+
+_`nimble-rich-text-editor`_
+
+![Editor](./spec-images/editor-sample.png)
+
+_`nimble-rich-text-viewer`_
+
+![Viewer](./spec-images/viewer-sample.png)
+
+## Design
+
+### `nimble-rich-text-editor`
+
+The `nimble-rich-text-editor` will be divided into two sections namely an `editor` section and a `footer` section.
+
+1. `editor` section is the actual text area to add or update rich text content.
+2. `footer` section consists of `nimble-toggle-button` to control each text formatting functionalities like bold, italic, etc, and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-button slot="footer-actions">Cancel</nimble-button>
+    <nimble-button slot="footer-actions">Add Comment</nimble-button>
+</nimble-rich-text-editor>
+```
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will serialize the content by extracting the Node from the editor and convert it into a markdown string using
+        [prosemirror-markdown serializer](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.ts#L30).
+    -   `setter` - this will parse the markdown string into a Node and load it back into the editor using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199).
+-   `isEmpty` - is a read-only property that indicates whether the editor is empty or not. This will be achieved through Tiptap's
```

Comment:
Not sure how critical it is to capture Angular implementation details in this spec, but I think it's at least worth mentioning how we would accomplish a bindable, read-only property in the Angular API. We would likely need to follow the pattern of the [`openChange` `Output` of the `NimbleBannerDirective`](https://github.com/ni/nimble/blob/1d5ab5f6f23e11e7b7c30600c196a04ea087c5af/angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts#L25), though without the need for the `Change` suffix, as there is no two-way binding concern. Doing this requires handling an event from the underlying component in order to emit the event from the directive (as done [here](https://github.com/ni/nimble/blob/1d5ab5f6f23e11e7b7c30600c196a04ea087c5af/angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts#L67) by the `NimbleBannerDirective`).

Reply:
For `isEmpty`, we have referred to the [read-only property `open`](https://github.com/ni/nimble/blob/1d5ab5f6f23e11e7b7c30600c196a04ea087c5af/packages/nimble-components/src/dialog/index.ts#L81) of the nimble dialog ([spec reference](https://github.com/ni/nimble/blob/1d5ab5f6f23e11e7b7c30600c196a04ea087c5af/packages/nimble-components/src/dialog/specs/README.md?plain=1#L63)) which returns true if the dialog is open. The Angular implementation of the `isEmpty` will be same as the [`open` attribute of the `NimbleDialogDirective`](https://github.com/ni/nimble/blob/1d5ab5f6f23e11e7b7c30600c196a04ea087c5af/angular-workspace/projects/ni/nimble-angular/src/directives/dialog/nimble-dialog.directive.ts#L51). The `isEmpty` getter method in the component simply returns the Tiptap's [isEmpty](https://tiptap.dev/api/editor#is-empty) value which will be true if the editor is empty.

With this approach, we can directly bind the `isEmpty` in the HTML of the Angular application. As an example, in the comments feature, we have the requirement to disable the `Comment` button when the editor is empty. So we can do something like the below code in the client to enable or disable the button.
```html
<nimble-rich-text-editor #editor>
    <nimble-button slot="footer-actions">Cancel</nimble-buttont>
    <nimble-button slot="footer-actions" [disabled]="editor.isEmpty">Comment</nimble-buttont>
</nimble-rich-text-editor>
```

So we thought of adding it as a read-only property rather than an event. We will update the spec to mention the Angular implementation for this API.

Reply:
Resolving this thread since I think it's a sufficient level of detail for the spec. @atmgrifter00 mentioned offline that he still wanted to understand this a bit better, but it doesn't need to block initial submission (we can always update the spec if our plans change).